src/components/visualizations/chart/highcharts/plugins/adjustTickAmount.ts

```diff
@@ -0,0 +1,246 @@
+        direction = getDirection(primaryAxis, axis)
+    ) {
+        let tickPositions: number[] = axis.tickPositions.slice();
+
+        if (direction === MOVE_ZERO_RIGHT) {
+            // add new tick to the start
+            tickPositions = addTick(tickPositions, tickInterval, true);
+            // remove last tick
+            tickPositions = tickPositions.slice(0, tickPositions.length - 1);
+        } else if (direction === MOVE_ZERO_LEFT) {
+            // add new tick to the end
+            tickPositions = addTick(tickPositions, tickInterval, false);
+            // remove first tick
+            tickPositions = tickPositions.slice(1, tickPositions.length);
+        }
+
+        axis.tickPositions = tickPositions;
+    }
+}
+
+function updateAxis(axis: IHighchartsAxisExtend, currentTickAmount: number): void {
+    const { options, tickPositions } = axis;
+
+    axis.transA *= (currentTickAmount - 1) / (Math.max(axis.tickAmount, 2) - 1); // avoid N/0 case
+
+    axis.min = options.startOnTick ? tickPositions[0] : Math.min(axis.min, tickPositions[0]);
+
+    axis.max = options.endOnTick
+        ? tickPositions[tickPositions.length - 1]
+        : Math.max(axis.max, tickPositions[tickPositions.length - 1]);
+}
+
+/**
+ * Prevent data is cut off by increasing tick interval to zoom out secondary axis
+ * Only apply to chart without user-input min/max
+ * @param axis
+ */
+export function preventDataCutOff(axis: IHighchartsAxisExtend): void {
+    const { chart } = axis;
+    const { min, max, dataMin, dataMax } = axis;
+
+    const isCutOff = !isUserSetExtremesOnAxes(chart) && (min > dataMin || max < dataMax);
+    if (!isCutOff) {
+        return;
+    }
+
+    axis.tickInterval *= 2;
+    axis.setTickPositions();
+}
+
+/**
+ * Copy and modify Highcharts behavior
+ */
+export function customAdjustTickAmount(): void {
+    const axis = this;
+    if (!axis.hasData()) {
+        return;
+    }
+
+    if (isYAxis(axis)) {
+        // persist tick amount value to calculate transA in 'updateAxis'
+        const currentTickAmount = (axis.tickPositions || []).length;
+        adjustTicks(axis);
+        alignSecondaryAxis(axis);
+        updateAxis(axis, currentTickAmount);
+        preventDataCutOff(axis);
+    }
+
+    // The finalTickAmt property is set in getTickAmount
+    const { finalTickAmt } = axis;
+    if (!isNil(finalTickAmt)) {
+        const len = axis.tickPositions.length;
+        let i = len;
+        while (i--) {
+            if (
+                // Remove every other tick
+                (finalTickAmt === 3 && i % 2 === 1) ||
+                // Remove all but first and last
+                (finalTickAmt <= 2 && i > 0 && i < len - 1)
+            ) {
+                axis.tickPositions.splice(i, 1);
+            }
+        }
+        axis.finalTickAmt = undefined;
+    }
+}
+
+/**
+ * Overwrite 'axis.adjustTickAmount' of Highcharts
+ * @param HighCharts
+ */
+export const adjustTickAmount = (HighCharts: any) => {
```

Comment:
(_Putting the comment here because I cannot add comments to diff images_)

The previous behaviour seemed better here, now it ends on 108% which is really strange. Is it something that could be fixed somehow? I'm afraid PM would not like it https://github.com/gooddata/gooddata-react-components/pull/1016/files?short_path=ce10dea#diff-ce10dea06729a9170f8538bf9287515f

This is also strange, bottom axis was nicely ending on 1000px but now ends on 1080, https://github.com/gooddata/gooddata-react-components/pull/1016/files?short_path=4e1c42a#diff-4e1c42aaf1b3d295934dd231cc84e91c

And this as well, Bigger amount axis max was 900 but now is like 1400 (so it's strange it did not pick 1000 as in other examples) https://github.com/gooddata/gooddata-react-components/pull/1016/files?short_path=35fecee#diff-35fecee1627b3e19b2819fe4f623c06f

Also, there are some changes which I don't know if they are good or bad, would be good to check with product management:
https://github.com/gooddata/gooddata-react-components/pull/1016/files?short_path=c0959e4#diff-c0959e4ef81213885f12c96fb0b5f6df (don't see any data and there are now ticks to -7)
https://github.com/gooddata/gooddata-react-components/pull/1016/files?short_path=b234a80#diff-b234a8066b5298e7b3718b38510f897c (added missing ticks)
https://github.com/gooddata/gooddata-react-components/pull/1016/files?short_path=257dc5e#diff-257dc5e428687d86483a69e164c3713e (more empty space in the last line chart than before)
https://github.com/gooddata/gooddata-react-components/pull/1016/files?short_path=c01449c#diff-c01449ccdcbe24f8cabd0ea57b72d76c (the middle bar chart starts 0 in the middle)

Reply:
@pbenes , plesae check my inline answer
> The previous behaviour seemed better here, now it ends on 108% which is really strange. Is it something that could be fixed somehow? I'm afraid PM would not like it https://github.com/gooddata/gooddata-react-components/pull/1016/files?short_path=ce10dea#diff-ce10dea06729a9170f8538bf9287515f
> 
> This is also strange, bottom axis was nicely ending on 1000px but now ends on 1080, https://github.com/gooddata/gooddata-react-components/pull/1016/files?short_path=4e1c42a#diff-4e1c42aaf1b3d295934dd231cc84e91c
> 
> And this as well, Bigger amount axis max was 900 but now is like 1400 (so it's strange it did not pick 1000 as in other examples) https://github.com/gooddata/gooddata-react-components/pull/1016/files?short_path=35fecee#diff-35fecee1627b3e19b2819fe4f623c06f
> 
For first 3 issues, we consider this an improvement. As you can compare before/after, tick lables are now less decimal than before. It's the main mission we want to achieve by leaving calculation and format to HC7


> Also, there are some changes which I don't know if they are good or bad, would be good to check with product management:
> https://github.com/gooddata/gooddata-react-components/pull/1016/files?short_path=c0959e4#diff-c0959e4ef81213885f12c96fb0b5f6df (don't see any data and there are now ticks to -7)
> 
For this case, I'm updating code to adapt by handling invalid min/max scale. We will hide its Y axis.

> https://github.com/gooddata/gooddata-react-components/pull/1016/files?short_path=b234a80#diff-b234a8066b5298e7b3718b38510f897c (added missing ticks)
I'm missing your point here, can you explain more?

> https://github.com/gooddata/gooddata-react-components/pull/1016/files?short_path=257dc5e#diff-257dc5e428687d86483a69e164c3713e (more empty space in the last line chart than before)
> https://github.com/gooddata/gooddata-react-components/pull/1016/files?short_path=c01449c#diff-c01449ccdcbe24f8cabd0ea57b72d76c (the middle bar chart starts 0 in the middle)
I will fix it soon :).

Reply:
Thanks for your replies, for the missing point I posted wrong link. I saw some charts with one value where before there was just one tick but now there are many ticks. I thought I had better example but unfortunately I cannot find better than this now (see the middle chart) https://github.com/gooddata/gooddata-react-components/pull/1016/files?short_path=6a22d4e#diff-6a22d4e8b42736b786ce212c25ef9193

Reply:
@pbenes , I have updated storybook images base on my new update.
Please check the diff in below link.
> Also, there are some changes which I don't know if they are good or bad, would be good to check with product management:
> https://github.com/gooddata/gooddata-react-components/pull/1016/files?short_path=c0959e4#diff-c0959e4ef81213885f12c96fb0b5f6df (don't see any data and there are now ticks to -7)
>
> https://github.com/gooddata/gooddata-react-components/pull/1016/files?short_path=257dc5e#diff-257dc5e428687d86483a69e164c3713e (more empty space in the last line chart than before)
> https://github.com/gooddata/gooddata-react-components/pull/1016/files?short_path=c01449c#diff-c01449ccdcbe24f8cabd0ea57b72d76c (the middle bar chart starts 0 in the middle)
> 

About this one, HC7 now moves the point from bottom from center.
Demo [HC7](https://jsfiddle.net/wc0x3dnj/) and [HC4](https://jsfiddle.net/6wahLb5e/)
And now I leave the ticks handle to HC7, so there is not just one tick anymore.
> I saw some charts with one value where before there was just one tick but now there are many ticks. I thought I had better example but unfortunately I cannot find better than this now (see the middle chart) https://github.com/gooddata/gooddata-react-components/pull/1016/files?short_path=6a22d4e#diff-6a22d4e8b42736b786ce212c25ef9193
>

In next commit, I will add some unit test for new update.